libs/core/src/lib/product-switch/product-switch-body/product-switch-body.component.ts

```diff
@@ -92,4 +108,67 @@ export class ProductSwitchBodyComponent implements OnInit {
             this.listMode = window.innerWidth < 776;
         }
     }
+
+    /** @hidden */
+    private handleNoListKeydown(event: KeyboardEvent): void {
+        const target = (<HTMLElement>event.target);
+        if (KeyUtil.isKey(event, 'ArrowLeft') && target.previousElementSibling) {
+            (<HTMLElement>target.previousElementSibling).focus();
+        } else if (KeyUtil.isKey(event, 'ArrowRight') && target.nextElementSibling) {
+            (<HTMLElement>target.nextElementSibling).focus();
+        } else if (KeyUtil.isKey(event, ['ArrowDown', 'ArrowUp'])) {
+            if (this.products.length >= 7) {
+                this.handleNoListMoreThanSeven(event);
+            } else if (this.products.length < 7) {
+                this.handleNoListLessThanSeven(event);
+            }
+        }
+    }
+
+    /** @hidden */
+    private handleNoListMoreThanSeven(event: KeyboardEvent): void {
+        const target = (<HTMLElement>event.target);
+        if (KeyUtil.isKey(event, 'ArrowDown')) {
+            if (target.nextElementSibling && target.nextElementSibling.nextElementSibling &&
+                target.nextElementSibling.nextElementSibling.nextElementSibling &&
+                target.nextElementSibling.nextElementSibling.nextElementSibling.nextElementSibling) {
+                (<HTMLElement>target.nextElementSibling.nextElementSibling.nextElementSibling.nextElementSibling).focus();
+            }
+        }
+        if (KeyUtil.isKey(event, 'ArrowUp')) {
+            if (target.previousElementSibling && target.previousElementSibling.previousElementSibling &&
+                target.previousElementSibling.previousElementSibling.previousElementSibling &&
+                target.previousElementSibling.previousElementSibling.previousElementSibling.previousElementSibling) {
+                (<HTMLElement>target.previousElementSibling.previousElementSibling.previousElementSibling
+                    .previousElementSibling).focus();
+            }
+        }
+    }
+
+    /** @hidden */
+    private handleNoListLessThanSeven(event: KeyboardEvent): void {
+        const target = (<HTMLElement>event.target);
+        if (KeyUtil.isKey(event, 'ArrowDown')) {
+            if (target.nextElementSibling && target.nextElementSibling.nextElementSibling &&
```

Comment:
What have you done 😆 
Please go recursive on that, or maybe there is a way to get all of the nodes of the parent and then easily get the desired node from the collection.

Reply:
Re: this vs. recursive, I think this is about as straightforward as it gets but ok